src/app/course/event/event-row/event-row.component.html

```diff
@@ -53,55 +53,69 @@
             Grade: {{ event.total_event_grade | number: '1.0-1' || 'TBA'}}%
         </ng-template>
     </div>
-    <div class="flex items-center">
+    <div class="questions-table__actions" *ngIf="event.has_edit_permission">
```

Comment:
The view button should not be under this permission. Please check the student view and make sure it works.